scripts/install-pytorch.sh

```diff
@@ -0,0 +1,174 @@
+      exit 0
+    else
+      echo "**** Current PyTorch commit $PYTORCH_CURRENT_COMMIT ****"
+      echo "**** Pinned PyTorch commit $PYTORCH_PINNED_COMMIT ****"
+    fi
+  fi
+  if [ "$FORCE_REINSTALL" = false ]; then
+    echo "**** Exiting without action. ****"
+    echo "**** INFO: Add --force-reinstall flag to force the PyTorch re-installation. ****"
+    exit 0
+  fi
+  pip uninstall -y torch
+fi
+
+############################################################################
+# Check installed torch pinned dependencies
+
+PINNED_TORCH_DEPENDENCIES_REGEX="^torchtext==|^torchaudio==|^torchvision=="
+INSTALLED_PINNED_TORCH_DEPENDENCIES=$(pip list --format=freeze | grep -iE "$PINNED_TORCH_DEPENDENCIES_REGEX" || true)
+
+if [ -n "$INSTALLED_PINNED_TORCH_DEPENDENCIES" ]; then
+  echo -e "**** The following pinned torch dependencies are installed: ****\n"
+  echo -e "$INSTALLED_PINNED_TORCH_DEPENDENCIES\n"
+  if [ "$FORCE_REINSTALL" = false ]; then
+    echo "**** Exiting without action. ****"
+    echo "**** INFO: Add --force-reinstall flag to force the PyTorch pinned dependencies re-installation. ****"
+    echo "**** INFO: PyTorch pinned dependencies build from source mode is not supported. ****"
+    exit 0
+  fi
+  pip uninstall -y torchtext torchaudio torchvision
+fi
+
+############################################################################
+# Install PyTorch and pinned dependencies from nightly builds.
+
+if [ "$BUILD_PYTORCH" = false ]; then
+  if `! which gh &> /dev/null` || `! which jq &> /dev/null`; then
+    echo "****** ERROR: gh or jq is missing. Please install gh and jq first. ******"
+    exit 1
+  fi
+  echo "**** Download nightly builds. ****"
+  PYTHON_VERSION=$( python -c "import sys; print(f'{sys.version_info.major}.{sys.version_info.minor}')" )
+  RUN_ID=$(gh run list -w "Triton wheels" -R intel/intel-xpu-backend-for-triton --json databaseId,conclusion | jq -r '[.[] | select(.conclusion=="success")][0].databaseId')
+  TEMP_DIR=$(mktemp -d)
+  WHEEL_PATTERN="wheels-pytorch-py${PYTHON_VERSION}*"
+  gh run download $RUN_ID \
+    --repo intel/intel-xpu-backend-for-triton \
+    --pattern "$WHEEL_PATTERN" \
+    --dir $TEMP_DIR
+  cd $TEMP_DIR/$WHEEL_PATTERN
+  echo "**** Install PyTorch and pinned dependencies from nightly builds. ****"
+  pip install torch*
+  rm -r $TEMP_DIR
+  exit 0
+fi
+
+############################################################################
+# Configure, build and install PyTorch from source.
+
+if [ ! -v BASE ]; then
+  echo "**** BASE is not given *****"
+  BASE=$ROOT/.scripts_cache
+  if [ ! -d "$BASE" ]; then
+    mkdir $BASE
+  fi
+  echo -e "**** Default BASE is set to $BASE ****\n"
+fi
+
+PYTORCH_PROJ=$BASE/pytorch
+SCRIPTS_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+
+echo "**** Cleaning $PYTORCH_PROJ before build ****"
+if rm -rf $PYTORCH_PROJ &>/dev/null; then
+  echo "**** $PYTORCH_PROJ have been cleaned ****"
+fi
```

Comment:
`rm -rf` always returns 0, no need in `if`.
```suggestion
rm -rf $PYTORCH_PROJ
```